### PR TITLE
WebGL2 support

### PR DESCRIPTION
**Pull request by eimink**
Quick and dirty support for WebGL2 shaders.

### CODE REVIEW

File: examples/test.html [hunk 1]
```diff
@@ -24,13 +24,21 @@
...
+	var webgl2 = parse("webgl2");
 	</script>
 
 	<script type="text/javascript" src="../shaderback.js"></script>
 	<script type="text/javascript">
 	if (shaderUri) {
 	  shaderback.setDebug(true);
-		window.onload = shaderback.loadURL(shaderUri);
+	    if(webgl2)
+		{
```
Comment (llewelld) [inline examples/test.html]: Two very minor points, but I'd be grateful if you could set the indentation to use tabs and the bracketing style to match for consistency:
```
if (a) {
}
else {
}
```

File: examples/triangles.txt [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    float m = S(0.03, 0.01, d);
+    float d2 =  length(a - b);
+    m *= S(1.2, 0.8, d2) * 0.5 + S(0.05, 0.03, abs(d2 - 0.75));
+    return m;
+}
+
+float N21(vec2 p){
+	p = fract(p * vec2(233.34, 851.73));
+    p += dot(p, p + 23.45);
+    return fract(p.x * p.y);
```
Comment (llewelld) [inline examples/triangles.txt]: This is a really nice example. Again, only a minor point, but if you could use either tabs or spaces here for the indentation, that would be appreciated.

File: shaderback.js [hunk 3]
```diff
@@ -105,11 +123,32 @@ var shaderback = (function () {
...
 
   // Initialise the OpenGL/WebGL canvas
-  function initGL(canvas) {
+  function initGL(canvas,webgl2) {
     var success = true;
     try {
-      gl = canvas.getContext("experimental-webgl");
-      gl.getExtension("OES_standard_derivatives");
+      if (webgl2)
+      {
```
Comment (llewelld) [inline shaderback.js]: Similarly for the brackets in this section.

**Review by llewelld — COMMENTED** [review]
This is a really nice addition and much appreciated. I just have a few style requests that I'd appreciate you looking at.

File: examples/test.html (hunk 1 above)
Comment (eimink) [reply examples/test.html]: Fixed in latest commit.

Comment (llewelld) [reply examples/test.html]: Marvellous, thank you, this is much appreciated!

**Comment by eimink** [comment]
Code indentation and brackets corrected now.

**Comment by llewelld** [comment]
> Code indentation and brackets corrected now.

Marvellous, thank you, this is much appreciated!

**Pull request merged**